FreeRTOS_IP.c

```diff
@@ -2251,8 +2251,17 @@ static eFrameProcessingResult_t prvProcessIPPacket( IPPacket_t * pxIPPacket,
                                 * these values. */
                                usLength = FreeRTOS_ntohs( pxUDPPacket->xUDPHeader.usLength );
 
-                               if( ( pxNetworkBuffer->xDataLength >= sizeof( UDPPacket_t ) ) &&
-                                   ( ( ( size_t ) usLength ) >= sizeof( UDPHeader_t ) ) )
+                               if( ( pxNetworkBuffer->xDataLength < sizeof( UDPPacket_t ) ) ||
+                                   ( ( ( size_t ) usLength ) < sizeof( UDPHeader_t ) ) )
+                               {
+                                   eReturn = eReleaseBuffer;
+                               }
+                               else if( usLength > ( FreeRTOS_ntohs( pxIPHeader->usLength ) - ipSIZE_OF_IPv4_HEADER ) )
+                               {
+                                   /* The UDP packet is bigger than the IP-payload. Something is wrong, drop the packet. */
+                                   eReturn = eReleaseBuffer;
+                               }
```

Comment:
Maybe one day we should add `iptrace` macros in all places where an invalid packet is dropped, but not now.